test.py

```diff
@@ -0,0 +1,69 @@
+import re
+import unittest
+from typing import NoReturn
+
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.remote.webelement import WebElement
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.common.exceptions import TimeoutException
+
+
+class HotlineTestCase(unittest.TestCase):
+    def setUp(self) -> NoReturn:
+        self.driver = webdriver.Firefox()
+
+    def get_with_wait(
+        self, xpath: str = None, selector: str = None, timeout: int = 10
+    ) -> WebElement:
+        if xpath is None and selector is None:
+            raise Exception("Find option is not defined")
+        try:
+            wait = WebDriverWait(self.driver, timeout)
+            locator = (
+                (By.XPATH, xpath) if xpath is not None
+                else (By.CSS_SELECTOR, selector)
+            )
+            return wait.until(EC.presence_of_element_located(locator))
+        except TimeoutException:
+            raise Exception("Could not find element")
+
+    def test_category_selection(self, category: str = "Книги") -> NoReturn:
+        self.driver.get("https://hotline.ua/")
+
+        category_link = self.get_with_wait(
+            selector=f'[data-menu-main-item="{category}"] > a'
+        )
+        category_link.click()
+
+        header = self.get_with_wait(selector=".cell-12 > h1")
+        assert header.text == category
+
+    def test_filter_by_option(self, option: str = "1920-2000") -> NoReturn:
+        self.driver.get("https://hotline.ua/computer/diski-ssd/")
+
+        option_link = self.get_with_wait(
+            xpath=f'//a[contains(text(), "{option}")]'
+        )
+        option_link.click()
+
+        label = option_link.get_attribute("data-eventlabel")
+        match = re.match(r'^.*\[(\d+)\]$', label)
```

Comment:
matching expected values using regular expressions introduces additional complexity

Reply:
I use regex to extract option id from string like this "option[id]". Could you explain why I shouldn't approach this task in such way?

Reply:
even if you really want to do that you should extract this action to a separate function. but still, I don't see a reason for that as e2e tests should interact with SUT in the same way as a user does, avoiding any kind of low-level interactions with DOM except when there is no other way to perform a required action

Reply:
Thank you. In this situation, I should check if an option is added to the selected list after a click on it. I can do that by matching strings stored in the 'innerText' attribute or using option id. What is the appropriate way to do that?

Reply:
you should use this approach only if you can't do this with webdriver .text attribute